src/db.rs

```diff
@@ -0,0 +1,154 @@
+pub async fn network_capacity(db: &Client) -> Result<Capacity> {
+    let records: Collection<Performance> = db
+        .database("host_statistics")
+        .collection("performance_summary");
+    let cursor = records.find(None, None).await?;
+
+    // cursor is a stream so it requires try_fold() from TryStreamExt
+    cursor
+        .try_fold(
+            Capacity {
+                total_hosts: 0,
+                read_only: 0,
+                source_chain: 0,
+            },
+            |mut total_capacity, performance| async move {
+                total_capacity.add_host(performance.uptime);
+                Ok(total_capacity)
+            },
+        )
+        .await
+        .map_err(Debug)
+}
+
+// Return all the hosts stored in `holoports_status` collection
+pub async fn list_all_hosts(db: &Client) -> Result<Vec<HostSummary>> {
+    // Retrieve and store in memory all holoport assignments
+    let hp_assignment: Collection<Assignment> = db
+        .database("host_statistics")
+        .collection("alpha_program_holoports");
+
+    let mut cursor = hp_assignment.find(None, None).await?;
+
+    let mut assignment_map = HashMap::new();
+
+    while let Some(a) = cursor.try_next().await? {
+        assignment_map.insert(a.name, "");
+    }
+    
+    // Retrieve all holoport statuses and format for an API response
+    let hp_status: Collection<Host> = db
+        .database("host_statistics")
+        .collection("holoports_status");
+
+    let pipeline = vec![
+        doc! {
+            // only successful ssh results
+            "$match": {
+                "sshSuccess": true
+            }
+        },
+        doc! {
+            // sort by timestamp, descending:
+            "$sort": {
+                "timestamp": -1
+            }
+        },
+        doc! {
+            "$group": {
+                "_id": "$name",
+                "IP": {"$first": "$IP"},
+                "timestamp": {"$first": "$timestamp"},
+                "sshSuccess": {"$first": "$sshSuccess"},
+                "holoNetwork": {"$first": "$holoNetwork"},
+                "channel": {"$first": "$channel"},
+                "holoportModel": {"$first": "$holoportModel"},
+                "hostingInfo": {"$first": "$hostingInfo"},
+                "error": {"$first": "$error"},
+            }
+        }
+    ];
+ 
+    let cursor= hp_status
+        .aggregate(pipeline, None)
+        .await?;
+    
+    // Update fields alpha_test and assigned_to based on the content of assignment_map
+    let cursor_extended = cursor.try_filter_map(|host| async {
+        println!("original host: {}", host);
+        let mut host: HostSummary = bson::from_document(host)?;
+        if let Some(assigned_to) = assignment_map.get(&host._id) {
+            host.alpha_program = Some(true);
+            host.assigned_to = Some(assigned_to.to_string());
+        }
+        
+        return Ok(Some(host));
```

Comment:
In rust the line in function without semi-colon at the end of the line (typically the last in block) is the return value of the function. Therefore using `return` keyword here is obsolete.
```suggestion
        Ok(Some(host))
```